Cray Fortran build errors with updated Fortran interface of SCORPIO

Steps to reproduce this issue on Frontier with Cray Clang compiler (not reproducible with GNU or AMD Clang)
```
. /usr/share/lmod/lmod/init/sh
module reset 
module switch PrgEnv-cray PrgEnv-cray/8.3.3
module switch cce cce/15.0.1
module switch craype craype/2.7.20

git clone https://github.com/E3SM-Project/scorpio.git
cd scorpio

mkdir build
cd build

CC=cc CXX=CC FC=ftn \
cmake -Wno-dev \
-DWITH_NETCDF=OFF \
-DPnetCDF_PATH=/opt/cray/pe/parallel-netcdf/1.12.3.1/crayclang/14.0 \
-DPIO_USE_MALLOC=ON \
..

make
```
Build errors
```
[ 76%] Building Fortran object src/flib/CMakeFiles/piof.dir/spio_inq_att.F90.o


ftn-855 ftn: ERROR SPIO_INQ_ATT, File = scorpio/src/flib/spio_inq_att.F90, Line = 13, Column = 8 
  The compiler has detected errors in module "SPIO_INQ_ATT".  No module information file will be created for this module.


ftn-878 ftn: WARNING SPIO_INQ_ATT, File = scorpio/src/flib/spio_inq_att.F90, Line = 20, Column = 7 
  A module named "ISO_C_BINDING" has already been directly or indirectly use associated into this scope.


ftn-1725 ftn: ERROR PIO_INQ_ATTNAME_FH_VID, File = scorpio/src/flib/spio_inq_att.F90, Line = 426, Column = 25 
  Unexpected syntax while parsing the WRITE statement : "EOS" was expected but found """.

Cray Fortran : Version 15.0.1 (20230120205242_66f7391d6a03cf932f321b9f6b1d8612ef5f362c)
Cray Fortran : Compile time:  0.0219 seconds
Cray Fortran : 1042 source lines
Cray Fortran : 2 errors, 1 warnings, 0 other messages, 0 ansi
Cray Fortran : "explain ftn-message number" gives more information about each message.
make[2]: *** [src/flib/CMakeFiles/piof.dir/build.make:162: src/flib/CMakeFiles/piof.dir/spio_inq_att.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:255: src/flib/CMakeFiles/piof.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

--- a/src/flib/spio_inq_att.F90
+++ b/src/flib/spio_inq_att.F90
@@ -422,7 +422,7 @@ CONTAINS
     file%iosystem => iosys
     
     IF((fh <= 0) .OR. (varid < 0) .OR. (attid <= 0)) THEN
-      WRITE(log_msg, *) "Invalid file handle/id (fh =", fh, ") "&
+      WRITE(log_msg, *) "Invalid file handle/id (fh =", fh, ") ",&
                         "or variable id (varid = ", varid,&
                         "or attribute id (attid = ", attid,&
                         ") passed to pio_inq_attname() function, pio_inq_attname_fh_vid()"

--- a/src/flib/spio_inq_file.F90
+++ b/src/flib/spio_inq_file.F90
@@ -377,7 +377,7 @@ CONTAINS
     IF(PRESENT(nDimensions)) THEN
       ierr = pio_inq_ndims(file, nDimensions)
       IF(ierr /= PIO_NOERR) THEN
-        WRITE(log_msg, *) "Unable to query the number of dimensions in file,"&
+        WRITE(log_msg, *) "Unable to query the number of dimensions in file,",&
                           "in function, pio_inquire_file(), file%fh = ", file%fh
         ierr = pio_error(file%iosystem, PIO_EINTERNAL, __PIO_FILE__, __LINE__, TRIM(log_msg))
         RETURN
@@ -387,7 +387,7 @@ CONTAINS
     IF(PRESENT(nVariables)) THEN
       ierr = pio_inq_nvars(file, nVariables)
       IF(ierr /= PIO_NOERR) THEN
-        WRITE(log_msg, *) "Unable to query the number of variables in file,"&
+        WRITE(log_msg, *) "Unable to query the number of variables in file,",&
                           "in function, pio_inquire_file(), file%fh = ", file%fh
         ierr = pio_error(file%iosystem, PIO_EINTERNAL, __PIO_FILE__, __LINE__, TRIM(log_msg))
         RETURN
@@ -397,7 +397,7 @@ CONTAINS
     IF(PRESENT(nAttributes)) THEN
       ierr = pio_inq_natts(file, nAttributes)
       IF(ierr /= PIO_NOERR) THEN
-        WRITE(log_msg, *) "Unable to query the number of attributes in file,"&
+        WRITE(log_msg, *) "Unable to query the number of attributes in file,",&
                           "in function, pio_inquire_file(), file%fh = ", file%fh
         ierr = pio_error(file%iosystem, PIO_EINTERNAL, __PIO_FILE__, __LINE__, TRIM(log_msg))
         RETURN
@@ -407,7 +407,7 @@ CONTAINS
     IF(PRESENT(unlimitedDimID)) THEN
       ierr = pio_inq_unlimdim(file, unlimitedDimID)
       IF(ierr /= PIO_NOERR) THEN
-        WRITE(log_msg, *) "Unable to query the unlimited dimension in file,"&
+        WRITE(log_msg, *) "Unable to query the unlimited dimension in file,",&
                           "in function, pio_inquire_file(), file%fh = ", file%fh
         ierr = pio_error(file%iosystem, PIO_EINTERNAL, __PIO_FILE__, __LINE__, TRIM(log_msg))
         RETURN

